modules/rostests/winetests/d3dx9_36/mesh.c

```diff
@@ -0,0 +1,11457 @@
+                case D3DFVF_XYZB1:
+                case D3DFVF_XYZB2:
+                case D3DFVF_XYZB3:
+                case D3DFVF_XYZB4:
+                case D3DFVF_XYZB5:
+                    pos_dim = (fvf & D3DFVF_POSITION_MASK) - D3DFVF_XYZB1 + 1;
+                    if (fvf & (D3DFVF_LASTBETA_UBYTE4 | D3DFVF_LASTBETA_D3DCOLOR))
+                    {
+                        pos_dim--;
+                        last_beta_dword = TRUE;
+                    }
+                    break;
+                case D3DFVF_XYZW: pos_dim = 4; break;
+            }
+            sprintf(prefix, "vertex[%u] position, ", i);
+            check_floats_(line, prefix, got_float, exp_float, pos_dim);
+            exp_float += pos_dim;
+            got_float += pos_dim;
+
+            if (last_beta_dword) {
+                ok_(__FILE__,line)(*(DWORD*)exp_float == *(DWORD*)got_float,
+                    "Vertex[%u]: Expected last beta %08x, got %08x\n", i, *(DWORD*)exp_float, *(DWORD*)got_float);
+                exp_float++;
+                got_float++;
+            }
+
+            if (fvf & D3DFVF_NORMAL) {
+                sprintf(prefix, "vertex[%u] normal, ", i);
+                check_floats_(line, prefix, got_float, exp_float, 3);
+                exp_float += 3;
+                got_float += 3;
+            }
+            if (fvf & D3DFVF_PSIZE) {
+                ok_(__FILE__,line)(compare(*exp_float, *got_float),
+                        "Vertex[%u]: Expected psize %g, got %g\n", i, *exp_float, *got_float);
+                exp_float++;
+                got_float++;
+            }
+            if (fvf & D3DFVF_DIFFUSE) {
+                ok_(__FILE__,line)(*(DWORD*)exp_float == *(DWORD*)got_float,
+                    "Vertex[%u]: Expected diffuse %08x, got %08x\n", i, *(DWORD*)exp_float, *(DWORD*)got_float);
+                exp_float++;
+                got_float++;
+            }
+            if (fvf & D3DFVF_SPECULAR) {
+                ok_(__FILE__,line)(*(DWORD*)exp_float == *(DWORD*)got_float,
+                    "Vertex[%u]: Expected specular %08x, got %08x\n", i, *(DWORD*)exp_float, *(DWORD*)got_float);
+                exp_float++;
+                got_float++;
+            }
+
+            texcount = (fvf & D3DFVF_TEXCOUNT_MASK) >> D3DFVF_TEXCOUNT_SHIFT;
+            for (j = 0; j < texcount; j++) {
+                DWORD dim = (((fvf >> (16 + 2 * j)) + 1) & 0x03) + 1;
+                sprintf(prefix, "vertex[%u] texture, ", i);
+                check_floats_(line, prefix, got_float, exp_float, dim);
+                exp_float += dim;
+                got_float += dim;
+            }
+
+            vertices = (BYTE*)vertices + vertex_size;
+            mesh_vertices = (BYTE*)mesh_vertices + vertex_size;
+        }
+    }
+
+    mesh->lpVtbl->UnlockVertexBuffer(mesh);
+}
+
+#define check_index_buffer(mesh, indices, num_indices, index_size) \
+    check_index_buffer_(__LINE__, mesh, indices, num_indices, 
```

Comment:
`{D3DXVECTOR3(0.0f,  3.0f,  0.0f), up, RED},` might work better for msvc

Reply:
I am still very confused about why it behaves differently on different architecture.